flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -771,22 +774,32 @@ PackagedProgram buildProgram(ProgramOptions options) throws FileNotFoundExceptio
 		String jarFilePath = options.getJarFilePath();
 		List<URL> classpaths = options.getClasspaths();
 
-		if (jarFilePath == null) {
-			throw new IllegalArgumentException("The program JAR file was not specified.");
+		// the job is not a specified a jar file or a Python job.
+		if (!(options.getJarFilePath() != null || options.isPython())) {
+			throw new IllegalArgumentException("The program should be specified a JAR file " +
+				"or a python file(or module)");
 		}
 
-		File jarFile = new File(jarFilePath);
+		File jarFile = null;
+		// If the job is specified a jar file
+		if (options.getJarFilePath() != null) {
+			jarFile = new File(jarFilePath);
 
-		// Check if JAR file exists
-		if (!jarFile.exists()) {
-			throw new FileNotFoundException("JAR file does not exist: " + jarFile);
-		}
-		else if (!jarFile.isFile()) {
-			throw new FileNotFoundException("JAR file is not a file: " + jarFile);
+			// Check if JAR file exists
+			if (!jarFile.exists()) {
+				throw new FileNotFoundException("JAR file does not exist: " + jarFile);
+			}
+			else if (!jarFile.isFile()) {
+				throw new FileNotFoundException("JAR file is not a file: " + jarFile);
+			}
```

Comment:
This section code only needs by the Java program, right?  If so, we should move this code into the above logic.